test/module/irohad/multi_sig_transactions/transport_test.cpp

```diff
@@ -27,26 +29,66 @@ using namespace shared_model::interface;
 
 using ::testing::_;
 using ::testing::A;
+using ::testing::DoAll;
 using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::SaveArg;
 
 class TransportTest : public ::testing::Test {
  public:
   TransportTest()
-      : async_call_(std::make_shared<AsyncGrpcClient<google::protobuf::Empty>>(
-            getTestLogger("AsyncClient"))),
-        parser_(std::make_shared<TransactionBatchParserImpl>()),
-        batch_validator_(
-            std::make_shared<shared_model::validation::BatchValidator>(
-                iroha::test::kTestsValidatorsConfig)),
-        batch_factory_(
-            std::make_shared<TransactionBatchFactoryImpl>(batch_validator_)),
-        tx_presence_cache_(
-            std::make_shared<iroha::ametsuchi::MockTxPresenceCache>()),
-        my_key_(makeKey()),
-        completer_(
-            std::make_shared<iroha::DefaultCompleter>(std::chrono::minutes(0))),
-        mst_notification_transport_(
-            std::make_shared<iroha::MockMstTransportNotification>()) {}
+      : my_key_(makeKey()),
+        stub(new iroha::network::transport::MockMstTransportGrpcStub()) {}
+  void SetUp() override {
+    async_call_ = std::make_shared<AsyncGrpcClient<google::protobuf::Empty>>(
+        getTestLogger("AsyncClient"));
+    parser_ = std::make_shared<TransactionBatchParserImpl>();
+    batch_validator_ =
+        std::make_shared<shared_model::validation::BatchValidator>(
+            iroha::test::kTestsValidatorsConfig);
+    batch_factory_ =
```

Comment:
just wondering, why did you move the initialization of some members from constructor to `SetUp()`?

Reply:
This change may protect someone who will unintentionally add side effect while writing new tests. This won't make the test run much longer but can save someone's time during test debugging.

Reply:
This change also helps you to reduce the amount of code since you don't need to duplicate the declaration of variables anymore. That is because SetUp() called automatically per each test.